ggml/src/ggml-cuda/ggml-cuda.cu

```diff
@@ -96,31 +96,22 @@ int ggml_cuda_get_device() {
 
 static cudaError_t ggml_cuda_device_malloc(void ** ptr, size_t size, int device) {
     ggml_cuda_set_device(device);
-#if defined(GGML_USE_HIP) && defined(GGML_HIP_UMA)
-    auto res = hipMallocManaged(ptr, size);
-    if (res == hipSuccess) {
-        // if error we "need" to know why...
-        CUDA_CHECK(hipMemAdvise(*ptr, size, hipMemAdviseSetCoarseGrain, device));
-    }
-    return res;
-#else
-
-#if !defined(GGML_USE_HIP)
     cudaError_t err;
     if (getenv("GGML_CUDA_ENABLE_UNIFIED_MEMORY") != nullptr)
     {
         err = cudaMallocManaged(ptr, size);
+#if defined(GGML_USE_HIP)
+        if (err == hipSuccess) {
+            // if error we "need" to know why...
```

Comment:
I know this was in the original code too, but i dont get what this comment is trying to tell us.

Reply:
CUDA_CHECK expands to ggml_cuda_error(). It prints the error details, line number and file name, then aborts the program. I think it meant more details could be seen in logs if `hipMemAdvise` went wrong.

Reply:
I don't understand the intended message of the comment either but I think the code is correct. Should we just remove the comment?

Reply:
Yes i think so.

Otherwise i like this pr and will approve once i have had the chance to test it locally.

Reply:
Removed.

Reply:
One wrinkle is that im not sure how cudaMallocManaged vs cudaMalloc behave on Nvidia apus (grace-hopper/ tegra) potentially the environment variable and what it implies dosent hold for those, in which case we should rename it back to HIP again and only apply it there.